fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionMgr.java

```diff
@@ -196,7 +198,31 @@ public static CompactionMgr loadCompactionManager(DataInput in) throws IOExcepti
         return compactionManager;
     }
 
+    public boolean isPartitionExist(PartitionIdentifier partition) {
+        Database db = GlobalStateMgr.getCurrentState().getDb(partition.getDbId());
+        if (db == null) {
+            return false;
+        }
+        db.readLock();
+        try {
+            // lake table or lake materialized view
+            OlapTable table = (OlapTable) db.getTable(partition.getTableId());
+            return table != null && table.getPartition(partition.getPartitionId()) != null;
+        } finally {
+            db.readUnlock();
+        }
+    }
+
     public void save(DataOutputStream dos) throws IOException, SRMetaBlockException {
```

Comment:
old persistence is used as default in 3.1
https://github.com/StarRocks/starrocks/blob/branch-3.1/fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java#L2048